quinn-proto/src/connection/mod.rs

```diff
@@ -214,6 +214,10 @@ where
     datagrams: DatagramState,
     /// Connection level statistics
     stats: ConnectionStats,
+    /// Supported versions.
+    supported_versions: Vec<u32>,
+    /// Initial version.
+    initial_version: u32,
```

Comment:
Style nits:
- Sentence fragments shouldn't end in periods.
- Comments should add useful information. If it just repeats the name of the variable, don't waste the space.

Also, I think we can replace both of these fields with
```rust
/// QUIC version used for the connection
version: u32
```
and pass `&[self.version]` in place of `&self.supported_versions`, as it's impossible for the version used by a connection to change.

Reply:
it seems like version negotiation is (or would be) part of the connection as it handles version negotiation packets. So I'm not sure about this change.

Reply:
did the change anyway, but I can revert it if you change your mind